src/main/scala/net/snowflake/spark/snowflake/pushdowns/querygeneration/MiscStatement.scala

```diff
@@ -104,7 +104,7 @@ private[querygeneration] object MiscStatement {
       case SortOrder(child, Descending, _, _) =>
         blockStatement(convertStatement(child, fields)) + "DESC"
 
-      case ScalarSubquery(subquery, _, _) =>
+      case ScalarSubquery(subquery, _, _, _) =>
```

Comment:
I saw your comments about `ScalarSubquery`. But, we can't `guess`. We have to make sure its usage.
If the new field is ignored, but it should not, it may cause wrong result. It is very difficult to diagnose this kind of problem.

Reply:
One more general comment is: Do you have a test case to test `Case` and `ScalarSubquery`?

Reply:
Regarding `ScalarSubquery`, I looked into it one more time.
Before the update it was:
`ScalarSubquery(plan, children, exprId)`, now it is `ScalarSubquery(plan, outerAttrs, exportId, joinCond)`.
Previously both, `outerAttrs` and `joinCond` references were stored on `children`. It was done so that the same field was not used for different purposes and to avoid confusion and has no effect on LogicalPlan itself.

Regarding tests.
Since I did not implement any changes other than a bit of tuning to updated interfaces, I was relying on existing tests, I saw there are some tests covering `CAST` and sub-queries.

If you agree with my suggestion about `CAST`(when ansi enabled) and `TRY_CAST` (by default), which I mentioned on my previous comment, I'll add a testcase for that.

Thanks again, for responding, and sorry for my `guesses`, let me know if my last comments make any sense, and we can move forward with this.

Reply:
Hey. Any plans to add this for the next 2.9.x release?